docs/README.md

```diff
@@ -70,41 +77,99 @@ root = "./app"
 
 ### Third-party modules
 
-Third-party modules are external Python packages that are not part of the standard library or your own project’s code. These are typically installed using a package manager like uv or pip and include libraries such as `requests`, `numpy`, or `django`.
+Third-party modules are Python packages that are not part of your project or the standard library.
+These are usually declared as dependencies in a `pyproject.toml` or `requirements.txt` file
+and installed using a package manager like uv or pip. Examples of popular third-party
+modules are `requests`, `numpy` and `django`.
+
+ty searches for third-party modules in the configured [Python environment](#python-environment).
+
+### Python environment
+
+The Python environment is used for discovery of third-party modules.
+
+By default, ty will attempt to discover a virtual environment.
+
+First, ty checks for an active virtual environment using the `VIRTUAL_ENV` environment variable. If
+not set, ty will search for a `.venv` directory in the project root or working directory. ty only
+supports discovery of virtual environments at this time.
```

Comment:
think it would still be great to mention that this is set by uv/poetry/hatch/pdm if they're running `uv run` or similar, and that it's set if you activate the environment explicitly in the shell

Reply:
Yeah working on that 👍

Reply:
i.e., https://github.com/astral-sh/ty/pull/335/commits/c97679219ab16ef725e3a0034f58e8b3a2996b71

Reply:
I think it sort of interrupts the flow more than it helps, but I don't feel strongly.

Reply:
I feel pretty strongly that this is important information -- it's part of the theme we're trying to describe here, which is "for most cases, you should (hopefully) not have to provide any configuration at all". If everybody starts setting `VIRTUAL_ENV` environment variables because they don't realise that uv is already going to set it for them (or, if they're using pip, because they don't realise it's already set when they activate the environment), that feels like a bad outcome? 😃

The note you added looks great; I'm happy with the text now!

Reply:
I agree it's important, but I think this is the wrong place for it. Those users need "guide"-style documentation that is explicitly targeting beginners (like we have in uv). This documentation _isn't_ for beginners. It's for people trying an experimental type checker. Catering to two audiences like that is just asking for confusing documentation. Anyway 🤷‍♀️

Reply:
that's all fair!